lib/ancestry.rb

```diff
@@ -6,4 +6,74 @@
 require_relative 'ancestry/materialized_path'
 
 module Ancestry
+  # used for materialized path
+  class MaterializedPathString < ActiveRecord::Type::Value
+    def initialize(casting: :to_i, delimiter: '/')
+      @casting = casting&.to_proc
+      @delimiter = delimiter
+    end
+
+    def type
+      :materialized_path_string
+    end
+
+    # convert to database type
+    def serialize(value)
+      if value.kind_of?(Array)
+        value.map(&:to_s).join(@delimiter).presence
+      elsif value.kind_of?(Integer)
+        value.to_s
+      elsif value.nil? || value.kind_of?(String)
+        value
+      else
+        byebug
```

Comment:
i think we want to handle this case a little differently

Reply:
lol